app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java

```diff
@@ -119,4 +182,77 @@ protected void onDestroy() {
         super.onDestroy();
         preferencesHelper.clear();
     }
+
+    void setUserDetails() {
+        LayoutInflater layoutInflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
+        navigationViewHeader = layoutInflater.inflate(R.layout.nav_drawer_header, navigationView);
+
+        {   //Set User name.
+            profileTextView = (TextView) navigationViewHeader.findViewById(R.id.tv_user_name);
+            profileTextView.setText(current_client.getDisplayName());
+            profileTextView.setVisibility(View.VISIBLE);
+        }
+        {   //Set User Image
+            profileImageView = (ImageView) navigationViewHeader.findViewById(R.id.Iv_user_image);
+            profileImageView.setImageResource(R.drawable.ic_material_user_icon_black_24dp);
+            setUserImage();
+        }
+    }
+
+
+    private void openUserDetails(Bitmap image) {
+        Intent profilePage = new Intent(HomeActivity.this, UserDetails.class);
+        profilePage.putExtra(Constants.CURRENT_CLIENT, current_client);
+        profilePage.putExtra(Constants.CLIENT_IMAGE, image);
+        startActivity(profilePage);
+    }
+
+    void setUserImage() {
+        {
+            final Call<ResponseBody> imageResponse = dataManager.getClientImage(clientId);
```

Comment:
You just broke the MVP architecure concept in HomeActivity. You have to make A HomeMvpView and HomePresenter and need to make logic separately.   

Please look other feature implementation